src/main/java/org/openpnp/model/Location.java

```diff
@@ -104,6 +104,10 @@ public double getLinearDistanceTo(double x, double y) {
 		return (Math.sqrt(Math.pow(this.x - x, 2) + Math.pow(this.y - y, 2)));
 	}
 	
+	public double getXyzDistanceTo(Location location) {
+		return (Math.sqrt(Math.pow(this.x - location.getX(), 2) + Math.pow(this.y - location.getY(), 2) + Math.pow(this.z - location.getZ(), 2)) );
```

Comment:
This should convert units before calculation, otherwise an incoming Location could be of a different unit and the results won't make sense. I realize that some other methods in this class have the same problem, so I am mostly putting this here as a note to myself to fix em all.